pytos/securetrack/helpers.py

```diff
@@ -1610,13 +1610,14 @@ def get_network_object_by_device_and_object_id(self, device_id, network_object_i
             response_string = self.get_uri(
                     "/securetrack/api/devices/{}/network_objects/{}".format(device_id, network_object_id),
                     expected_status_codes=200).response.content
-            network_object = Network_Objects_List.from_xml_string(response_string)[0]
+            network_object = [network_object for network_object in Network_Objects_List.from_xml_string(response_string)
```

Comment:
This is so wrong as it will result in bug. There is no validation that object ID and passed to function ID are of the same type. Meaning that if you search for device with ID "2" - it will fail though the list will contain this device.
I do understand that you took the same logic as in bug with name, but it is not valid here:
1) There is always only one object with same ID, otherwise there is bug in ST
2) ST just always return list and not a single object, but it will be always only one object with same ID or no objects at all.

Reply:
The function doc defines the network_object_id param as an int, so if you pass "2", you're not using the function properly.

Your statements 1 and 2 are not correct. The API does not always return a list with a single object. if the the passed object has members. It will return objects for the members as well. It doesn't always return them in a consistent order either, so returning index 0 can result in the bug in #8